### PR TITLE
Fix benchmark tests

### PR DESCRIPTION
**Pull request by tonyredondo**
## Summary of changes

This PR fixes the benchmark tests

### CODE REVIEW

**Review by andrewlock — APPROVED** [review]
Thanks :D

File: tracer/build/_build/Build.cs [hunk 1]
```diff
@@ -412,11 +412,11 @@ void DeleteReparsePoints(string path)
...
                 );
 
-            var (framework, runtimes) = IsOsx switch
-            {
-                true => (TargetFramework.NETCOREAPP3_1, "net6.0"),
-                false => (TargetFramework.NET6_0, "net472 netcoreapp3.1.0"),
-            };
+                var (framework, runtimes) = IsOsx switch
+                {
+                    true => (TargetFramework.NETCOREAPP3_1, "net6.0"),
```
Comment (andrewlock) [inline tracer/build/_build/Build.cs:417]: wait, is this right 🤔 
```suggestion
                    true => (TargetFramework.NETCOREAPP3_1, "netcoreapp3.1"),
```

Comment (tonyredondo) [reply tracer/build/_build/Build.cs:417]: yes because there's no .NET Core 3.1 for OSX (at least for ARM64) version

Comment (tonyredondo) [reply tracer/build/_build/Build.cs:417]: I'll no click in you suggestions. :trollface:

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #4210 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/2d0c4418b5714ab43f05ec1d5edf296d97fe1b67): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|668μs|202ns|782ns|0| 0|0|2.63&nbsp;KB|
|#4210|`WriteAndFlushEnrichedTraces`|net472|795μs|166ns|621ns|0.396| 0|0|3.31&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`AllCycleSimpleBody`|netcoreapp3.1|41.4μs|66.8ns|259ns|0.0206| 0|0|1.63&nbsp;KB|
|#4210|`AllCycleSimpleBody`|net472|42.5μs|14.2ns|55.1ns|0.252| 0|0|1.69&nbsp;KB|
|#4210|`AllCycleMoreComplexBody`|netcoreapp3.1|232μs|417ns|1.56μs|0.115| 0|0|9.12&nbsp;KB|
|#4210|`AllCycleMoreComplexBody`|net472|240μs|239ns|861ns|1.45| 0|0|9.28&nbsp;KB|
|#4210|`ObjectExtractorSimpleBody`|netcoreapp3.1|173ns|0.0756ns|0.273ns|0.00366| 0|0|272&nbsp;B|
|#4210|`ObjectExtractorSimpleBody`|net472|146ns|0.119ns|0.445ns|0.0446| 0|0|281&nbsp;B|
|#4210|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|4.07μs|2.07ns|7.76ns|0.0508| 0|0|3.78&nbsp;KB|
|#4210|`ObjectExtractorMoreComplexBody`|net472|4.15μs|4.76ns|18.4ns|0.618| 0.00629|0|3.89&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`SendRequest`|netcoreapp3.1|182μs|212ns|820ns|0.184| 0|0|19.92&nbsp;KB|
|#4210|`SendRequest`|net472|0.00107ns|0.000363ns|0.00141ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|624μs|2.01μs|7.78μs|0.302| 0|0|42.09&nbsp;KB|
|#4210|`WriteAndFlushEnrichedTraces`|net472|784μs|3.83μs|15.3μs|8.33| 2.65|0.379|53.24&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`ExecuteNonQuery`|netcoreapp3.1|1.14μs|0.267ns|1ns|0.00971| 0|0|728&nbsp;B|
|#4210|`ExecuteNonQuery`|net472|1.32μs|0.782ns|2.93ns|0.11| 0.00066|0|690&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`CallElasticsearch`|netcoreapp3.1|1.17μs|0.344ns|1.29ns|0.0101| 0|0|752&nbsp;B|
|#4210|`CallElasticsearch`|net472|1.93μs|1.18ns|4.25ns|0.12| 0.000956|0|762&nbsp;B|
|#4210|`CallElasticsearchAsync`|netcoreapp3.1|1.22μs|1.73ns|6.46ns|0.0109| 0|0|800&nbsp;B|
|#4210|`CallElasticsearchAsync`|net472|2.04μs|0.971ns|3.76ns|0.13| 0.00101|0|819&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`ExecuteAsync`|netcoreapp3.1|1.38μs|0.66ns|2.47ns|0.0124| 0|0|920&nbsp;B|
|#4210|`ExecuteAsync`|net472|1.53μs|0.579ns|2.24ns|0.14| 0.000762|0|883&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`SendAsync`|netcoreapp3.1|4.2μs|2.45ns|9.48ns|0.0294| 0|0|2.26&nbsp;KB|
|#4210|`SendAsync`|net472|6.76μs|1.98ns|7.66ns|0.445| 0|0|2.82&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`EnrichedLog`|netcoreapp3.1|2.03μs|0.951ns|3.68ns|0.0222| 0|0|1.63&nbsp;KB|
|#4210|`EnrichedLog`|net472|2.3μs|1.81ns|6.51ns|0.245| 0|0|1.55&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`EnrichedLog`|netcoreapp3.1|118μs|160ns|620ns|0| 0|0|4.22&nbsp;KB|
|#4210|`EnrichedLog`|net472|150μs|133ns|514ns|0.672| 0.224|0|4.39&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`EnrichedLog`|netcoreapp3.1|3.67μs|0.925ns|3.34ns|0.0494| 0|0|3.7&nbsp;KB|
|#4210|`EnrichedLog`|net472|4.74μs|1.5ns|5.42ns|0.527| 0.00237|0|3.32&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`SendReceive`|netcoreapp3.1|1.5μs|0.812ns|3.15ns|0.0135| 0|0|1.01&nbsp;KB|
|#4210|`SendReceive`|net472|1.79μs|0.861ns|3.22ns|0.16| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`EnrichedLog`|netcoreapp3.1|3.53μs|1.01ns|3.64ns|0.0211| 0|0|1.58&nbsp;KB|
|#4210|`EnrichedLog`|net472|4.08μs|1.36ns|5.08ns|0.312| 0|0|1.97&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`StartFinishSpan`|netcoreapp3.1|541ns|0.308ns|1.19ns|0.00753| 0|0|544&nbsp;B|
|#4210|`StartFinishSpan`|net472|645ns|0.279ns|1.08ns|0.0865| 0|0|546&nbsp;B|
|#4210|`StartFinishScope`|netcoreapp3.1|730ns|0.243ns|0.877ns|0.00919| 0|0|664&nbsp;B|
|#4210|`StartFinishScope`|net472|799ns|0.893ns|3.46ns|0.0993| 0|0|626&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|#4210|`RunOnMethodBegin`|netcoreapp3.1|818ns|0.394ns|1.42ns|0.00911| 0|0|664&nbsp;B|
|#4210|`RunOnMethodBegin`|net472|972ns|0.579ns|2.17ns|0.0993| 0|0|626&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (4210)](https://github.com/DataDog/dd-trace-dotnet/tree/7dda5233910f21b25299ca1886efc258cad04cda)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/51b25d6a27a8b1d264f26b3fde567510c85efb0d)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (3,038ms)  : 2934, 3142
     .   : milestone, 3038,
    master - mean (3,054ms)  : 2974, 3133
     .   : milestone, 3054,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (3,789ms)  : 3686, 3891
     .   : milestone, 3789,
    master - mean (3,791ms)  : 3667, 3916
     .   : milestone, 3791,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (3,135ms)  : 2973, 3296
     .   : milestone, 3135,
    master - mean (3,127ms)  : 3013, 3241
     .   : milestone, 3127,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (3,600ms)  : 3553, 3647
     .   : milestone, 3600,
    master - mean (3,596ms)  : 3544, 3649
     .   : milestone, 3596,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (3,141ms)  : 3035, 3246
     .   : milestone, 3141,
    master - mean (3,138ms)  : 3027, 3248
     .   : milestone, 3138,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (3,594ms)  : 3498, 3690
     .   : milestone, 3594,
    master - mean (3,579ms)  : 3520, 3638
     .   : milestone, 3579,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (188ms)  : 184, 191
     .   : milestone, 188,
    master - mean (188ms)  : 184, 191
     .   : milestone, 188,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (1,049ms)  : 1017, 1080
     .   : milestone, 1049,
    master - mean (1,054ms)  : 1018, 1089
     .   : milestone, 1054,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (367ms)  : 362, 373
     .   : milestone, 367,
    master - mean (368ms)  : 362, 374
     .   : milestone, 368,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (1,142ms)  : 1108, 1176
     .   : milestone, 1142,
    master - mean (1,144ms)  : 1120, 1167
     .   : milestone, 1144,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4210) - mean (355ms)  : 348, 363
     .   : milestone, 355,
    master - mean (357ms)  : 350, 363
     .   : milestone, 357,

    section CallTarget+Inlining+NGEN
    This PR (4210) - mean (1,103ms)  : 1076, 1129
     .   : milestone, 1103,
    master - mean (1,101ms)  : 1075, 1127
     .   : milestone, 1101,

```

**Pull request merged**